include/sparrow/buffer/dynamic_bitset.hpp

```diff
@@ -388,45 +424,95 @@ namespace sparrow
     }
 
     template <random_access_range B>
-    auto dynamic_bitset_base<B>::begin() -> iterator
+    constexpr auto dynamic_bitset_base<B>::begin() -> iterator
     {
         return iterator(this, data(), 0u);
     }
 
     template <random_access_range B>
-    auto dynamic_bitset_base<B>::end() -> iterator
+    constexpr auto dynamic_bitset_base<B>::end() -> iterator
     {
         block_type* block = m_buffer.size() ? data() + m_buffer.size() - 1 : data();
         return iterator(this, block, size() % s_bits_per_block);
     }
 
     template <random_access_range B>
-    auto dynamic_bitset_base<B>::begin() const -> const_iterator
+    constexpr auto dynamic_bitset_base<B>::begin() const -> const_iterator
     {
         return cbegin();
     }
 
     template <random_access_range B>
-    auto dynamic_bitset_base<B>::end() const -> const_iterator
+    constexpr auto dynamic_bitset_base<B>::end() const -> const_iterator
     {
         return cend();
     }
 
     template <random_access_range B>
-    auto dynamic_bitset_base<B>::cbegin() const -> const_iterator
+    constexpr auto dynamic_bitset_base<B>::cbegin() const -> const_iterator
     {
         return const_iterator(this, data(), 0u);
     }
 
     template <random_access_range B>
-    auto dynamic_bitset_base<B>::cend() const -> const_iterator
+    constexpr auto dynamic_bitset_base<B>::cend() const -> const_iterator
     {
         const block_type* block = m_buffer.size() ? data() + m_buffer.size() - 1 : data();
         return const_iterator(this, block, size() % s_bits_per_block);
     }
 
     template <random_access_range B>
-    dynamic_bitset_base<B>::dynamic_bitset_base(storage_type&& buf, size_type size)
+    constexpr auto dynamic_bitset_base<B>::at(size_type pos) -> reference
+    {
+        if (pos >= size())
+        {
+            throw std::out_of_range(
+                "dynamic_bitset_base::at: index out of range for dynamic_bitset_base of size"
+                + std::to_string(size()) + " at index " + std::to_string(pos)
+            );
+        }
+        return (*this)[pos];
+    }
+
+    template <random_access_range B>
+    constexpr auto dynamic_bitset_base<B>::at(size_type pos) const -> const_reference
+    {
+        if (pos >= size())
+        {
+            throw std::out_of_range(
+                "dynamic_bitset_base::at: index out of range for dynamic_bitset_base of size"
+                + std::to_string(size()) + " at index " + std::to_string(pos)
+            );
+        }
+        return test(pos);
+    }
+
+    template <random_access_range B>
+    constexpr auto dynamic_bitset_base<B>::front() -> reference
+    {
```

Comment:
Add assertions in non-throwing functions that access the elements (front, back, etc.)?

Reply:
done